src/main/native/GCM.c

```diff
@@ -1141,16 +1141,16 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
 #ifdef DEBUG_GCM_DATA
     if ( debug ) {
         gslogMessagePrefix ("DATA_GCM ivNative : ");
-        gslogMessageHex (ivNative, 0, (int) ivLen, 0, 0, NULL);
+        gslogMessageHex ((char*) ivNative, 0, (int) ivLen, 0, 0, NULL);
```

Comment:
shouldnt we also try to get the message for `(char*)iv` instead of `(char*)ivNative`?

Reply:
On line 1126 `iv` is set to `ivNative` so at this point in the code I think they are the same value. This being said we are also debugging and making use of the variable `ivNative` on line 1167 below so it seems most accurate to print the value of `ivNative` here. Unless there is a different path I am missing where printing the value of `iv` is helpful?

Reply:
makes sense